### ISSUE TITLE
Using as_text() on a physical key InputEvent returns an empty string silently

### ISSUE THREAD

**Issue by Isonzo** [godotengine/godot#61508]
### Godot version

3.4.4

### System information

Linux, Debian 11

### Issue description

As of now, when you call the `as_text` method in a physical key `InputEvent`, it simply returns an empty string. I would expect it to return either an error telling the user that the method does not function with physical keys, or for it to return the key in that physical position.

All in all, I realize this is a very small issue, and you can currently fix it by simply using keys instead of physical keys, but I think it should do something other than silently return nothing. It should be more obvious something is wrong.

### Steps to reproduce

1. Make a new Godot project
2. Add an action in the Input Map, in my case I called it test_input
3. Assign a physical key to that action, in my case I chose A
4. Create a Control Node 
5. Add a script to the Control Node

```
extends Control


func _ready():
	var event: InputEvent = InputMap.get_action_list("test_input")[0]
	var faulty_as_text: String = event.as_text()
	print(faulty_as_text)
```
6. Save and run the scene

It will not print anything, and if you add a breakpoint, you will see that although there is an `InputEvent` Object in event, but `faulty_as_text` remains empty. It will do this silently.

### Minimal reproduction project

[InputEventBug.zip](https://github.com/godotengine/godot/files/8791711/InputEventBug.zip)

This project simply follows the above instructions. Simply run it and behold.

**Comment by quantumedbox** [comment]
As far as I see problematic piece of code is this:
```c++
// core/os/input_event.cpp
String InputEventKey::as_text() const {
	String kc = keycode_get_string(scancode);
	if (kc == String()) {
		return kc;
	}
```
Events registered with physical keys don't have scancode member assigned from InputMap, so, `keycode_get_string` just returns nullptr
I don't know precisely how distinction is made internally, but something like this does seem to work:
```c++
uint32_t code = physical_scancode ? physical_scancode : scancode;
String kc = keycode_get_string(code);
```

**Comment by linkpy** [comment]
The issue is no longer present on 4a11 :

![image](https://user-images.githubusercontent.com/2700596/177002266-eae813fb-dd32-426a-8c13-a35a3559c7bc.png)

The commit that modified `InputEventKey::as_text` is dfe4c5f8e13, so maybe we could just backport this commit to 3.x?

**Comment by akien-mga** [comment]
Fixed by #62633.
